### DISCUSSION TITLE
Compiling EMGU on Nvidia Xavier with Cuda failing

### DISCUSSION THREAD

**Discussion by ofer** [emgucv/emgucv#776] [Q&A]
Hello folks,

I've been trying to compile EMGU on an Nvidia Xavier for the past week and I just am not able to figure out how to set this up correctly.

Without Cuda, the latest mainline compiles fine, but when I enable cuda, the following occures when in the opencv build phase:
[  0%] Built target opencv_highgui_plugins
[  1%] Built target libopenjp2
[  3%] Built target carotene_objs
[  4%] Built target tegra_hal
[  4%] Built target opencv_videoio_plugins
[  4%] Built target opencv_cudev
[  4%] Built target opencv_core_pch_dephelp
[  4%] Built target pch_Generate_opencv_core
[  5%] Built target zlib
[  5%] Built target ittnotify
[  9%] Built target opencv_core
[  9%] Built target opencv_cudaarithm_pch_dephelp
[  9%] Built target pch_Generate_opencv_cudaarithm
[ 10%] Built target opencv_cudaarithm
[ 11%] Built target opencv_flann_pch_dephelp
[ 12%] Built target pch_Generate_opencv_flann
[ 12%] Built target opencv_flann
[ 12%] Built target opencv_hdf_pch_dephelp
[ 12%] Built target pch_Generate_opencv_hdf
Scanning dependencies of target opencv_hdf
[ 12%] Building CXX object opencv/modules/hdf/CMakeFiles/opencv_hdf.dir/src/hdf5.cpp.o
[ 12%] Linking CXX shared library ../../../bin/runtimes/ubuntu-arm64/native/libopencv_hdf.so
/usr/bin/ld: ../../../install/lib/libhdf5.a(H5PLint.c.o): in function `H5PL__close':
H5PLint.c:(.text+0x640): undefined reference to `dlclose'
/usr/bin/ld: ../../../install/lib/libhdf5.a(H5PLint.c.o): in function `H5PL__open':
H5PLint.c:(.text+0x6b0): undefined reference to `dlopen'
/usr/bin/ld: H5PLint.c:(.text+0x6c4): undefined reference to `dlsym'
/usr/bin/ld: H5PLint.c:(.text+0x6e0): undefined reference to `dlsym'
/usr/bin/ld: H5PLint.c:(.text+0x784): undefined reference to `dlerror'
/usr/bin/ld: ../../../install/lib/libhdf5.a(H5PLplugin_cache.c.o): in function `H5PL__find_plugin_in_cache':
H5PLplugin_cache.c:(.text+0x398): undefined reference to `dlsym'
collect2: error: ld returned 1 exit status
make[2]: *** [opencv/modules/hdf/CMakeFiles/opencv_hdf.dir/build.make:89: bin/runtimes/ubuntu-arm64/native/libopencv_hdf.so.4.6.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:6185: opencv/modules/hdf/CMakeFiles/opencv_hdf.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

I've read on the internet that the build system isn't adding the dl library to the linking dependencies (-ldl) but i can't for the life of me figure out how to add it in.  I tried defining the following in my environment:

LDFLAGS="-ldl"
OPENCV_EXTRA_EXE_LINKER_FLAGS="-ldl"
OPENCV_EXTRA_MODULE_LINKER_FLAGS="-ldl"
OPENCV_EXTRA_SHARED_LINKER_FLAGS="-ldl"

None of this works, anybody have any ideas? 
This is compiling the latest version of emgu master:
34efc17e3844bab028f2b49e51cdf854b1f5833e

on a Xavier with the latest jetpack (basically meaning ubuntu 20.04.5 on an arm64 platform and cuda 11.4)

**Comment by emgucv** [comment]
Looks like a problem with the HDF5 module on your system. If you don't use the HDF5, you can add the CMake flag: 

-DBUILD_opencv_hdf:BOOL=OFF 

to disable opencv hdf module. That should work around that error message.

**Comment by ofer** [comment]
Excellent, I re-used this idea to "solve" the other issues that occured, in total, to compile using CUDA on an nVidia Xavier, i disabled the following:

      -DBUILD_opencv_hdf:BOOL=FALSE \
      -DBUILD_opencv_viz:BOOL=FALSE \
      -DBUILD_opencv_cudalegacy:BOOL=FALSE \

I hope I don't end up needing these parts.
